tools/src/main/scala/org/apache/spark/sql/rapids/tool/AppFilterImpl.scala

```diff
@@ -87,8 +92,26 @@ class AppFilterImpl(
     } else {
       apps
     }
+
+   // If `any` is specified as logicFilter config, then store the value in variable which is used
+   // later for final computation.
+   val appNameFilteredForDisjunction = if (appArgs.applicationName.isSupplied &&
+       filterAppName.nonEmpty && appArgs.any()
+       && logicFilter.contains("any")) {
```

Comment:
I think the `appArgs.applicationName.isSupplied` check is redundant here because we really just care about `filterAppName.nonEmpty` and it can only be non-empty if supplied? The same applies to some other similar checks.

Reply:
Yes, you are right. Not sure why I had both checks :) . Removed the redundant check.